src/constfold.d

```diff
@@ -971,41 +971,125 @@ extern (C++) UnionExp Cmp(TOK op, Loc loc, Type type, Expression e1, Expression
             rawCmp = cast(int)(es1.len - es2.len);
         n = specificCmp(op, rawCmp);
     }
-    else if (e1.isConst() != 1 || e2.isConst() != 1)
-    {
-        emplaceExp!(CTFEExp)(&ue, TOKcantexp);
-        return ue;
-    }
-    else if (e1.type.isreal())
-    {
-        r1 = e1.toReal();
-        r2 = e2.toReal();
-        goto L1;
-    }
-    else if (e1.type.isimaginary())
-    {
-        r1 = e1.toImaginary();
-        r2 = e2.toImaginary();
-    L1:
-        n = realCmp(op, r1, r2);
-    }
-    else if (e1.type.iscomplex())
-    {
-        assert(0);
-    }
-    else
+    else if (e1.isConst() == 1 && e2.isConst() == 1)
```

Comment:
It does not.  It returns `2` for some stuff that isn't const but can be accessed in ctfe.